RecoLocalCalo/EcalRecProducers/plugins/EcalUncalibRecHitPhase2WeightsProducer.cc

```diff
@@ -0,0 +1,109 @@
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+
+#include "DataFormats/EcalDigi/interface/EcalDigiCollections.h"
+#include "DataFormats/EcalRecHit/interface/EcalRecHitCollections.h"
+#include "DataFormats/EcalRecHit/interface/EcalUncalibratedRecHit.h"
+
+#include "DataFormats/EcalDigi/interface/EcalConstants.h"
+
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+
+class EcalUncalibRecHitPhase2WeightsProducer : public edm::stream::EDProducer<> {
+public:
+  explicit EcalUncalibRecHitPhase2WeightsProducer(const edm::ParameterSet& ps);
+  void produce(edm::Event& evt, const edm::EventSetup& es) override;
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  float tRise_;
+  float tFall_;
+  std::vector<double> weights_;
+
+  edm::EDGetTokenT<EBDigiCollectionPh2> ebDigiCollectionInToken_;
+  edm::EDPutTokenT<EBUncalibratedRecHitCollection> ebDigiCollectionOutToken_;
```

Comment:
Maybe better to call it `ebUncalibRecHitCollectionOutToken_` instead or something along this line. There are no ebDigis produced after all.